modules/standard/Random.chpl

```diff
@@ -667,11 +843,21 @@ module Random {
         //if arr.domain.type.strideable then
         //  compilerError("Shuffle requires non-strideable 1-D array");
 
+        if parSafe then
```

Comment:
In case you didn't notice, it is also available as a standalone routine (around line 2091). I don't have a strong opinion about where exactly it goes, but this is a good thing to talk about later.